src/pages/RealmExplorer.js

```diff
@@ -3,201 +3,214 @@
+    case 'date':
+      return parseDate(value);
+    default:
+      return '';
   }
+};
 
-  onFilterChange(filterString) {
-    const { dataSource } = this.state;
-
-    if (!this.unfilteredData) return;
-
-    let data = null;
-    if (filterString === '') {
-      // if filter is emptied, clear filter.
-      data = this.unfilteredData;
-    } else {
-      try {
-        // Use |this.unfilteredData| to avoid stacking filters.
-        data = this.unfilteredData.filtered(filterString);
-      } catch (err) {
-        // Ignore error silently.
-      }
-    }
+const getInitialState = database => {
+  const objectString = 'Requisition';
+  const objectData = database.objects(objectString);
+  const searchString = '';
+  const filterString = '';
+  const filteredData = objectData;
+  return { objectString, objectData, searchString, filterString, filteredData };
+};
 
-    if (data) {
-      this.setState({
-        data,
-        dataSource: dataSource.cloneWithRows(data),
-      });
+const getUpdatedState = (database, state) => {
+  const { objectString, objectData, searchString, filterString, filteredData } = state;
+  const newState = { searchString, filterString };
+  const updateObject = searchString !== objectString && OBJECT_TYPES.indexOf(searchString) >= 0;
+  newState.objectString = updateObject ? searchString : objectString;
+  newState.objectData = updateObject ? database.objects(newState.objectString) : objectData;
+  if (filterString === '') {
+    newState.filteredData = newState.objectData;
+  } else {
+    try {
+      newState.filteredData = newState.objectData.filtered(filterString);
+    } catch (err) {
+      newState.filteredData = updateObject ? newState.objectData : filteredData;
     }
   }
 
-  onSearchChange(searchTerm) {
-    const { database } = this.props;
-    const { dataSource } = this.state;
+  return newState;
+};
 
-    if (OBJECT_TYPES.indexOf(searchTerm) < 0) return;
-    const data = database.objects(searchTerm);
-    this.unfilteredData = data;
-    this.setState({
-      data,
-      dataSource: dataSource.cloneWithRows(data),
-    });
-  }
+/**
+ * A page for displaying objects in the local database. Includes search and filtering functionality.
+ *
+ * @prop   {UIDatabase}     database       App wide database.
+ * @state  {string}         objectString   Current database object.
+ * @state  {string}         searchString   Current search string. Used to update current object.
+ * @state  {string}         filterString   Current filter string. Used to update filtered data.
+ * @state  {Realm.Results}  objectData     Reference to current database object results. Used to
+ *                                         roll back filter state when filter is reset.
+ * @state  {Realm.Results}  filteredData   Reference to current database object results after filter
+ *                                         has been applied. Displayed to the user.
+ */
+export const RealmExplorer = ({ database }) => {
```

Comment:
just import `UIDatabase` so we can remove passing of database object through props

Reply:
c9d5f59d202b2c892b933fefee9017dad994524e.